packages/host/tests/integration/components/ai-assistant-panel/codeblocks-test.gts

```diff
@@ -279,4 +279,291 @@ module('Integration | ai-assistant-panel | codeblocks', function (hooks) {
+
+    simulateRemoteMessage(
+      roomId,
+      '@aibot:localhost',
+      {
+        body: messageWithNestedPreTags,
+        formatted_body: messageWithNestedPreTags,
+        msgtype: 'org.text',
+        format: 'org.matrix.custom.html',
+        isStreamingFinished: true,
+      },
+      {
+        origin_server_ts: new Date(2024, 0, 3, 12, 30).getTime(),
+      },
+    );
+
+    await waitFor('[data-test-message-idx="0"]');
+
+    assert
+      .dom('.monaco-editor')
+      .exists({ count: 2 }, 'Should have 2 monaco editors');
+
+    let monacoContent = getMonacoContent();
+    assert.ok(
+      monacoContent.includes('Basic pre tag example'),
+      'Monaco content includes the first comment',
+    );
+    assert.ok(
+      monacoContent.includes('This is preformatted text'),
+      'Monaco content includes the first nested pre content',
+    );
+    assert.ok(
+      monacoContent.includes('function sayHello()'),
+      'Monaco content includes the code block',
+    );
+    assert.ok(
+      monacoContent.includes('const data = {'),
+      'Monaco content includes the styled pre block',
+    );
+    assert.equal(
+      (document.getElementsByClassName('view-lines')[1] as HTMLElement)
+        .innerText,
+      '// existing code ... \nlet a = 1;\nlet c = 3;\n// new code ... \nlet a = 2;',
+    );
+
+    assert.dom('ol li').exists({ count: 4 }, 'Should have 4 list items');
+
+    await percySnapshot(assert);
+  });
+
+  test('handles HTML tags outside backticks', async function (assert) {
+    let roomId = await renderAiAssistantPanel(`${testRealmURL}Person/fadhlan`);
+    let messageWithHtmlOutsideBackticks = `Here's some HTML outside of code blocks:
+
+<p>This is a paragraph with <strong>bold text</strong> and <em>italic text</em>.</p>
+
+<ul>
+  <li>List item 1</li>
+  <li>List item 2 with <a href="https://example.com">a link</a></li>
+</ul>
+
+<div class="container">
+  <h1>Heading 1</h1>
+  <p>Another paragraph with <code>inline code</code>.</p>
+</div>
+
+And here's a code block with HTML inside:
+
+\`\`\`html
+<div class="example">
+  <p>This HTML is inside a code block with language specified.</p>
+</div>
+\`\`\`
+
+And another code block without language specified:
+
+\`\`\`
+<div class="example">
+  <p>This HTML is inside a code block without language specified.</p>
+</div>
+\`\`\``;
+
+    simulateRemoteMessage(
+      roomId,
+      '@aibot:localhost',
+      {
+        body: messageWithHtmlOutsideBackticks,
+        formatted_body: messageWithHtmlOutsideBackticks,
+        msgtype: 'org.text',
+        format: 'org.matrix.custom.html',
+        isStreamingFinished: true,
+      },
+      {
+        origin_server_ts: new Date(2024, 0, 3, 12, 30).getTime(),
+      },
+    );
+
+    await waitFor('[data-test-message-idx="0"]');
+
+    // Check that HTML outside backticks is displayed as actual HTML
+    assert
+      .dom('[data-test-ai-message-content]')
+      .containsText(
```

Comment:
Same here, I think it's a clearer and better test if you assert the whole content string compared to many `containsText()`s